AdobeAnaltyicsAPI.gs

```diff
@@ -128,6 +158,67 @@ var AdobeAnalyticsAPIController = (function() {
 
   };
 
+
+  /**
+   * AdobeAnalyticsAPIController.prototype.getJwtToken()
+   * get JSON Web Token
+   *
+   * @return {String} JSON Web Token
+   **/
+  AdobeAnalyticsAPIController.prototype.getJwtToken = function () {
+  
+    this.Logger("", {
+      "title": "getJwtToekn"
+    });
+
+    const header = Utilities.base64Encode(JSON.stringify({"alg":"RS256","typ":"JWT"}), Utilities.Charset.UTF_8);
+    const myJwtPayload = JSON.parse(JWT_PAYLOAD);
+    myJwtPayload.exp = Math.floor(Date.now() / 1000 + 600) // expire after 10 minutes (600 sec)
+    const claimSet = JSON.stringify(myJwtPayload);
+    const encodeText = header + "." + Utilities.base64Encode(claimSet, Utilities.Charset.UTF_8).slice(0, -2);
+    const signature = Utilities.computeRsaSha256Signature(encodeText, PRIVATE_KEY);
+    const jwtToken = encodeText + "." + Utilities.base64Encode(signature).slice(0, -2);
+    this.Logger(jwtToken, {
+      "prefix": "JWT Token"
+    });
+
+    return jwtToken;
+  
+  };
+  
+  /**
+   * AdobeAnalyticsAPIController.prototype.getAccessToken()
+   * get Analytics Report API Access Token
+   *
+   * @param {String} JSON Web Token
+   * @return {String} Access Token
+   **/
+  AdobeAnalyticsAPIController.prototype.getAccessToken = function(jwtToken) {
+
+    this.Logger(jwtToken, {
+      "title": "getAccessToken"
+    });
+
+    const url = 'https://ims-na1.adobelogin.com/ims/exchange/jwt';
+    const payload = {
+      "client_id" : API_KEY,
+      "client_secret" : CLIENT_SECRET,
+//      "jwt_token" : Utilities.base64Encode(jwtToken, Utilities.Charset.UTF_8)
```

Comment:
please remove comment if you don't need.
(I think it's not necessary. so do you think about it?)

Reply:
I will remove the line since it's not necessary. Thank you for your comment.